CRM/Contact/BAO/GroupContact.php

```diff
@@ -211,6 +213,13 @@ public static function removeContactsFromGroup(
           'tracking' => $tracking,
         ];
         CRM_Contact_BAO_SubscriptionHistory::create($historyParams);
+        // Removing a row from civicrm_group_contact for a smart group may mean a contact
+        // Is now back in a group based on criteria so we will invalidate the cache if it is there
+        // So that accurate group cache is created next time it is needed.
+        $groupInformation = civicrm_api3('Group', 'getsingle', ['id' => $groupId]);
+        if (!empty($groupInformation['saved_search_id'])) {
+          CRM_Contact_BAO_GroupContactCache::invalidateGroupContactCache($groupId);
```

Comment:
Actually don't check for saved search id here - it is also relevant if a group has children and I don't think it would be BAD to do if not relevant

Reply:
@eileenmcnaughton agree